app/server/services/idamService.ts

```diff
@@ -0,0 +1,71 @@
+import * as request from 'superagent';
+require('superagent-proxy')(request);
+const config = require('config');
+
+
+const appPort = config.get('node.port');
+
+const apiUrl = config.get('idam.api-url');
+const appSecret = config.get('idam.client.secret');
+const httpProxy = config.get('httpProxy');
+
+interface TokenResponse {
+  access_token: string
+}
+
+interface UserDetails {
+  email: string
+}
+
+async function getToken(code: string, protocol: string, host: string): Promise<TokenResponse> {
+  try {
+    const redirect_uri: string = getRedirectUrl(protocol, host);
+
+    const response: request.Response = await makeProxiedRequest(request.post(`${apiUrl}/oauth2/token`) as ProxyRequest)
+      .auth('sscs-cor', appSecret)
+      .set('Accept', 'application/json')
+      .set('Content-Type', 'application/x-www-form-urlencoded')
+      .type('form')
+      .send({"grant_type": 'authorization_code'})
+      .send({"code": code})
+      .send({"redirect_uri": redirect_uri});
+
+    return Promise.resolve(response.body);
+  } catch (error) {
+    console.log("failed to make call" + error);
+    return Promise.reject(error);
+  }
+}
+
+async function getUserDetails(token: string): Promise<UserDetails> {
+  try {
+    const response: request.Response = await makeProxiedRequest(request.get(`${apiUrl}/details`) as ProxyRequest)
+      .set('Accept', 'application/json')
+      .set('Authorization', 'Bearer ' + token);
+
+    return Promise.resolve(response.body);
+  } catch (error) {
+    return Promise.reject(error);
```

Comment:
Please track this error in app insights too. We've been explicitly tracking all errors resulting from calls out to other services/APIs etc.